homework_2/hw2.py

```diff
@@ -0,0 +1,159 @@
+def gc_consist(seq):
+    
+    gc_count = seq.count('G') + seq.count('C') 
+    gc_percentage = gc_count/ (len(seq)-1) * 100
+    #minus 1 because of '\n'
+    
+    return gc_percentage
+
+
+def score(string):
+    
+    score_all = []
+    summ = 0
+    
+    for symbol in string: #count the quality for each symbol in the string
+        
+        score_symbol = ord(symbol) - 33
+        score_all.append(score_symbol)
+    
+    mean = sum(score_all) / (len(score_all)-1) #count mean
+    #minus 1 because of '\n'
+    
+    return mean
+
+
+def filter_gc(file, gc_bounds):
+    
+    filter_list_gc = []
+    nofilter_list = []
+    
+    with open(file, 'r') as f:
+    
+        while True:
+            
+            #getting information about one sequence
+            one_seq = []
+
+            for i in range(4):
+                one_seq.append(f.readline())
+
+            #check the end of file
+            if len(one_seq[1]) == 0:
+
+                break
+            
+            #filterring by gc
+            elif (gc_consist(one_seq[1]) <= gc_bounds[1] and
+                  gc_consist(one_seq[1]) >= gc_bounds[0]):
```

Comment:
Лучше не вызывать два раза функцию, которую можно вызвать 1 раз. И ещё можно записать выражение чуть проще
```suggestion
            gc_content = gc_consist(one_seq[1])   # Это должно быть вне условия
            elif gc_bounds[0] <= gc_content <= gc_bounds[1]:
```